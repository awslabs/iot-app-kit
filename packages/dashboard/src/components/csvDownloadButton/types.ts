import { Primitive } from '@iot-app-kit/core';

export type CSVDownloadObject = {
  id: string;
  timestamp: string;
  value?: Primitive;
  unit?: string;
  aggregationType?: string;
  resolution?: number;
  propertyName?: string;
  assetName?: string;
  propertyAlias?: string;
  assetId?: string;
  dataType?: string;
  propertyId?: string;
};
