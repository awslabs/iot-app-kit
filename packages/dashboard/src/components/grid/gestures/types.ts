import type { Position } from '~/types';
import { type RegisteredWidgetType } from '~/features/widget-plugins/registry';

export interface DragEvent {
  target?: EventTarget;
  start: Position;
  end: Position;
  vector: Position;
  union: boolean;
}

export interface PointClickEvent {
  target?: EventTarget;
  position: Position;
  union: boolean;
}

export interface DropEvent<WidgetType extends RegisteredWidgetType> {
  position: Position;
  item: {
    widgetType: WidgetType;
  };
}
