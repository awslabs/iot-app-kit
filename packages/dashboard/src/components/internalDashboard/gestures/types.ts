export type Gesture = 'move' | 'resize' | 'select' | undefined;
