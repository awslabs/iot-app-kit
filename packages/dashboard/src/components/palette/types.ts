import { ComponentTag } from '~/types';

export type ComponentPaletteDraggable = {
  componentTag: ComponentTag;
  rect: DOMRect | null;
};
