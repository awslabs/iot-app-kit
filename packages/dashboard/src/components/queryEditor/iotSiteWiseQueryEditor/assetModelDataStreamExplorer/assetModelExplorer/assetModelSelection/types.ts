export type AssetModel = { id: string; name: string };
