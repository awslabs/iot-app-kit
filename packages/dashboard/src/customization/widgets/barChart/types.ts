import type { AggregateType } from '@aws-sdk/client-iotsitewise';
import type { ThresholdSettings } from '@iot-app-kit/core';
import type { QueryProperties } from '~/types/queries';
import type { AxisSettings, ThresholdWithId } from '../../settings';
import type { BarChartWidgetType } from './constants';

declare module '~/types/widgets' {
  interface DashboardWidgetRegistry {
    [BarChartWidgetType]: BarChartProperties;
  }
}

export interface BarChartProperties extends QueryProperties {
  title?: string;
  thresholds?: ThresholdWithId[];
  thresholdSettings?: ThresholdSettings;
  axis?: AxisSettings;
  significantDigits?: number;
  resolution?: string;
  aggregationType?: AggregateType;
}
