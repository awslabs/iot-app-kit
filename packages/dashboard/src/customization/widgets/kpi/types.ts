import type { StyledThreshold } from '@iot-app-kit/core';
import type { AssistantProperty } from '@iot-app-kit/react-components';
import type { SimpleFontSettings } from '~/customization/settings';
import type { QueryProperties } from '~/types/queries';
import type { KPIWidgetType } from './constants';

declare module '~/types/widgets' {
  interface DashboardWidgetRegistry {
    [KPIWidgetType]: KPIProperties;
  }
}

export interface KPIProperties extends QueryProperties {
  title?: string;
  primaryFont: SimpleFontSettings;
  secondaryFont: SimpleFontSettings;
  showAggregationAndResolution?: boolean;
  showValue?: boolean;
  showUnit?: boolean;
  showIcon?: boolean;
  showName?: boolean;
  showTimestamp?: boolean;
  showDataQuality?: boolean;
  thresholds?: StyledThreshold[];
  backgroundColor?: string;
  significantDigits?: number;
  assistant?: AssistantProperty;
}
