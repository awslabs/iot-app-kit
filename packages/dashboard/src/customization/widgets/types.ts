import type { StyleSettingsMap } from '@iot-app-kit/core';
import type { SiteWiseAssetQuery } from '@iot-app-kit/source-iotsitewise';
import type { Widget } from '~/types';
import type {
  AxisSettings,
  ComplexFontSettings,
  LegendSettings,
  SimpleFontSettings,
  ThresholdSettings,
} from '../settings';
import type { TableColumnDefinition, TableItem } from '@iot-app-kit/react-components/src';

export type QueryConfig<S, T> = {
  source: S;
  query: T;
};

export type SiteWiseQueryConfig = QueryConfig<'iotsitewise', SiteWiseAssetQuery | undefined>;

export type QueryProperties = {
  styleSettings?: StyleSettingsMap;
  queryConfig: SiteWiseQueryConfig;
};

export type KPIProperties = QueryProperties & {
  primaryFont: SimpleFontSettings;
  secondaryFont: SimpleFontSettings;
  showValue?: boolean;
  showUnit?: boolean;
  showIcon?: boolean;
  showName?: boolean;
  showTimestamp?: boolean;
  thresholdSettings?: ThresholdSettings;
};

export type StatusProperties = QueryProperties & {
  primaryFont: SimpleFontSettings;
  secondaryFont: SimpleFontSettings;
  showValue?: boolean;
  showUnit?: boolean;
  showIcon?: boolean;
  showName?: boolean;
  thresholdSettings?: ThresholdSettings;
  backgroundColor?: string;
};

export type LineChartProperties = QueryProperties & {
  thresholdSettings?: ThresholdSettings;
  axis?: AxisSettings;
  legend?: LegendSettings;
};

export type ScatterChartProperties = QueryProperties & {
  thresholdSettings?: ThresholdSettings;
  axis?: AxisSettings;
  legend?: LegendSettings;
};

export type BarChartProperties = QueryProperties & {
  thresholdSettings?: ThresholdSettings;
  axis?: AxisSettings;
  legend?: LegendSettings;
};

export type TableProperties = QueryProperties & {
  thresholdSettings?: ThresholdSettings;
  fontSettings?: ComplexFontSettings;
  items?: TableItem[];
  columnDefinitions?: TableColumnDefinition[];
};

export type TextProperties = {
  fontSettings?: ComplexFontSettings;
  value: string;
  isUrl?: boolean;
  href?: string;
};

export type QueryWidget = Widget<QueryProperties>;

export type KPIWidget = Widget<KPIProperties>;
export type StatusWidget = Widget<StatusProperties>;
export type LineChartWidget = Widget<LineChartProperties>;
export type ScatterChartWidget = Widget<ScatterChartProperties>;
export type BarChartWidget = Widget<BarChartProperties>;
export type TableWidget = Widget<TableProperties>;
export type TextWidget = Widget<TextProperties>;
