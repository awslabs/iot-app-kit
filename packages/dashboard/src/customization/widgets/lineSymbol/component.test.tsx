import React from 'react';
import { render } from '@testing-library/react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import LineWidgetComponent from './component';
import { Provider } from 'react-redux';
import { configureDashboardStore } from '~/store';
import { DashboardState } from '~/store/state';
import { LineProperties, LineWidget } from '../types';

const mockWidgetPartial = {
  id: 'mock-line-widget',
  type: 'line-symbol',
  x: 0,
  y: 0,
  z: 1,
  width: 7,
  height: 4,
};

const TestComponent = (object: { widget: LineWidget; isSelected: boolean }) => {
  const { widget, isSelected } = object;

  const initialState: Partial<DashboardState> = {
    dashboardConfiguration: {
      widgets: [widget],
    },
    selectedWidgets: isSelected ? [widget] : [],
  };

  return (
    <Provider store={configureDashboardStore(initialState)}>
      <DndProvider backend={HTML5Backend}>
        <LineWidgetComponent {...widget} />
      </DndProvider>
    </Provider>
  );
};

describe('LineWidgetComponent', () => {
  describe('Rendering', () => {
    [
      {
        properties: {
          lineStyle: 'solid',
          color: 'black',
          thickness: 5,
          start: { x: 20, y: 20 },
          end: { x: 180, y: 180 },
        } as LineProperties,
        isSelected: true,
      },
      {
        properties: {
          lineStyle: 'dashed',
          color: 'black',
          thickness: 5,
          start: { x: 30, y: 20 },
          end: { x: 180, y: 190 },
        } as LineProperties,
        isSelected: true,
      },
      {
        properties: {
          lineStyle: 'dotted',
          color: 'black',
          thickness: 5,
          start: { x: 25, y: 5 },
          end: { x: 100, y: 120 },
        } as LineProperties,
        isSelected: true,
      },
      {
        properties: {
          lineStyle: 'solid',
          color: 'red',
          thickness: 5,
          start: { x: 20, y: 20 },
          end: { x: 180, y: 180 },
        } as LineProperties,
        isSelected: true,
      },
      {
        properties: {
          lineStyle: 'dashed',
          color: 'blue',
          thickness: 10,
          start: { x: 180, y: 175 },
          end: { x: 20, y: 30 },
        } as LineProperties,
        isSelected: true,
      },
      {
        properties: {
          lineStyle: 'dotted',
          color: 'green',
          thickness: 15,
          start: { x: 180, y: 175 },
          end: { x: 20, y: 30 },
        } as LineProperties,
        isSelected: true,
      },
    ].forEach((configuration) => {
      const widget = {
        ...mockWidgetPartial,
        properties: { ...configuration.properties },
      };
      const isSelected = configuration.isSelected;
      it(`should render correctly with ${JSON.stringify(configuration)} while ${
        isSelected ? 'selected' : 'not selected'
      }`, () => {
        const { container } = render(
          <TestComponent widget={widget} isSelected={isSelected} />
        );
        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('Anchor visibility', () => {
    const widget = {
      ...mockWidgetPartial,
      properties: {
        lineStyle: 'solid',
        color: 'black',
        thickness: 5,
        start: { x: 180, y: 175 },
        end: { x: 20, y: 30 },
      },
    } as LineWidget;

    it('should display anchors when the widget is selected', async () => {
      const { getAllByTestId } = render(
        <TestComponent widget={widget} isSelected={true} />
      );
      expect(getAllByTestId('line-anchor')).toHaveLength(2);
    });
    it('should not display anchors when the widget is not selected', () => {
      const { queryAllByTestId } = render(
        <TestComponent widget={widget} isSelected={false} />
      );
      expect(queryAllByTestId('line-anchor')).toHaveLength(0);
    });
  });
});
