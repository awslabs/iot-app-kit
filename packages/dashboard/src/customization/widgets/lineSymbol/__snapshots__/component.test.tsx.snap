// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineWidgetComponent Rendering should render correctly with {"properties":{"lineStyle":"dashed","color":"black","thickness":5,"start":{"x":30,"y":20},"end":{"x":180,"y":190}},"isSelected":true} while selected 1`] = `
<div>
  <div
    class="fit-full"
  >
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 20px; top: 10px;"
    />
    <svg
      class="fit-full"
      pointer-events="none"
    >
      <line
        stroke="black"
        stroke-dasharray="10,5"
        stroke-width="5"
        x1="30"
        x2="180"
        y1="20"
        y2="190"
      />
    </svg>
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 170px; top: 180px;"
    />
  </div>
</div>
`;

exports[`LineWidgetComponent Rendering should render correctly with {"properties":{"lineStyle":"dashed","color":"blue","thickness":10,"start":{"x":180,"y":175},"end":{"x":20,"y":30}},"isSelected":true} while selected 1`] = `
<div>
  <div
    class="fit-full"
  >
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 170px; top: 165px;"
    />
    <svg
      class="fit-full"
      pointer-events="none"
    >
      <line
        stroke="blue"
        stroke-dasharray="10,5"
        stroke-width="10"
        x1="180"
        x2="20"
        y1="175"
        y2="30"
      />
    </svg>
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 10px; top: 20px;"
    />
  </div>
</div>
`;

exports[`LineWidgetComponent Rendering should render correctly with {"properties":{"lineStyle":"dotted","color":"black","thickness":5,"start":{"x":25,"y":5},"end":{"x":100,"y":120}},"isSelected":true} while selected 1`] = `
<div>
  <div
    class="fit-full"
  >
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 15px; top: -5px;"
    />
    <svg
      class="fit-full"
      pointer-events="none"
    >
      <line
        stroke="black"
        stroke-dasharray="3,3"
        stroke-width="5"
        x1="25"
        x2="100"
        y1="5"
        y2="120"
      />
    </svg>
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 90px; top: 110px;"
    />
  </div>
</div>
`;

exports[`LineWidgetComponent Rendering should render correctly with {"properties":{"lineStyle":"dotted","color":"green","thickness":15,"start":{"x":180,"y":175},"end":{"x":20,"y":30}},"isSelected":true} while selected 1`] = `
<div>
  <div
    class="fit-full"
  >
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 170px; top: 165px;"
    />
    <svg
      class="fit-full"
      pointer-events="none"
    >
      <line
        stroke="green"
        stroke-dasharray="3,3"
        stroke-width="15"
        x1="180"
        x2="20"
        y1="175"
        y2="30"
      />
    </svg>
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 10px; top: 20px;"
    />
  </div>
</div>
`;

exports[`LineWidgetComponent Rendering should render correctly with {"properties":{"lineStyle":"solid","color":"black","thickness":5,"start":{"x":20,"y":20},"end":{"x":180,"y":180}},"isSelected":true} while selected 1`] = `
<div>
  <div
    class="fit-full"
  >
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 10px; top: 10px;"
    />
    <svg
      class="fit-full"
      pointer-events="none"
    >
      <line
        stroke="black"
        stroke-dasharray="1,0"
        stroke-width="5"
        x1="20"
        x2="180"
        y1="20"
        y2="180"
      />
    </svg>
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 170px; top: 170px;"
    />
  </div>
</div>
`;

exports[`LineWidgetComponent Rendering should render correctly with {"properties":{"lineStyle":"solid","color":"red","thickness":5,"start":{"x":20,"y":20},"end":{"x":180,"y":180}},"isSelected":true} while selected 1`] = `
<div>
  <div
    class="fit-full"
  >
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 10px; top: 10px;"
    />
    <svg
      class="fit-full"
      pointer-events="none"
    >
      <line
        stroke="red"
        stroke-dasharray="1,0"
        stroke-width="5"
        x1="20"
        x2="180"
        y1="20"
        y2="180"
      />
    </svg>
    <div
      class="line-anchor"
      data-testid="line-anchor"
      draggable="true"
      style="position: absolute; left: 170px; top: 170px;"
    />
  </div>
</div>
`;
